x-pack/elastic-agent/pkg/capabilities/input.go

```diff
@@ -0,0 +1,271 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package capabilities
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/transpiler"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/status"
+)
+
+const (
+	inputsKey = "inputs"
+)
+
+func newInputsCapability(log *logger.Logger, rd ruleDefinitions, reporter status.Reporter) (Capability, error) {
+	caps := make([]Capability, 0, len(rd))
+
+	for _, r := range rd {
+		c, err := newInputCapability(log, r, reporter)
+		if err != nil {
+			return nil, err
+		}
+
+		if c != nil {
+			caps = append(caps, c)
+		}
+	}
+
+	return &multiInputsCapability{log: log, caps: caps}, nil
+}
+
+func newInputCapability(log *logger.Logger, r ruler, reporter status.Reporter) (Capability, error) {
+	cap, ok := r.(*inputCapability)
+	if !ok {
+		return nil, nil
+	}
+
+	cap.log = log
+	cap.reporter = reporter
+	return cap, nil
+}
+
+type inputCapability struct {
+	log      *logger.Logger
+	reporter status.Reporter
+	Name     string `json:"name,omitempty" yaml:"name,omitempty"`
+	Type     string `json:"rule" yaml:"rule"`
```

Comment:
Variable name and json block do not seem to be aligned?

Reply:
intentional i have a Rule fn to match the interface, this needs to be exposed so i went with Type